VideoWeb/VideoWeb/ClientApp/src/app/shared/venue-list/vh-officer-venue-list/vh-officer-venue-list.component.ts

```diff
@@ -68,4 +94,22 @@ export class VhOfficerVenueListComponent extends VenueListComponentDirective imp
     get showVhoSpecificContent(): boolean {
         return true;
     }
+
+    async updateCsoFilterSelection(filter: CsoFilter) {
+        const selectCso = (csoId: string) => {
+            this.selectedCsos = [...this.selectedCsos, csoId];
+        };
```

Comment:
is this function selectCso neccessary? Would doing `this.selectedCsos.push(VhOfficerVenueListComponent.ALLOCATED_TO_ME)` 
`this.selectedCsos.push(id)` 
`this.selectedCsos.push(VhOfficerVenueListComponent.UNALLOCATED)`

on lines 106 and 109 and 112 not achieve the same thing?

Reply:
thanks. Yes that's a good shout. I probably over-abstracted it with that function

Reply:
I've had to revert this change as it broke the pre-populating of the dropdown list when navigating back to the page.

I remember why I did it that way now - doing .push does not trigger change detection for the dropdown list - https://github.com/ng-select/ng-select#change-detection